src/main/java/Duke.java

```diff
@@ -1,10 +1,163 @@
+                isBye = true;
+            } else if (line.equalsIgnoreCase("list")) {
+                requestList(tasks);
+            } else if (line.contains("done")) {
+                inputDone(line, tasks);
+            } else if (line.contains("undo")) {
+                undoDone(line, tasks);
+            } else if (isValidTaskInput(line)) {
+                taskManager(line, tasks);
+            } else {
+                inputErrorMessage();
+            }
+        }
+    }
+
+    public static boolean isValidTaskInput(String input) {
+        if (input.toLowerCase().contains("todo") || input.toLowerCase().contains("deadline") || input.toLowerCase().contains("event")) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public static void taskManagerMessage(Task[] tasks) {
+        printHorizontalLine();
+        System.out.println("Understood. I've added this task:");
+        System.out.println(tasks[taskCount]);
+        System.out.println("Now you have " + taskCount + " tasks in the list.");
+        printHorizontalLine();
+    }
+
+    public static void taskManager(String input, Task[] tasks) {
+        int dividePos = input.trim().indexOf(" ");
+        String taskType = input.trim().substring(0, dividePos).toLowerCase();
+
+        if (taskType.equalsIgnoreCase("todo")) {
+            addTodo(input, tasks);
+        } else if (taskType.equalsIgnoreCase("deadline") || taskType.equalsIgnoreCase("event")) {
+            addDeadlineOrEvent(input, tasks);
+        }
+        taskManagerMessage(tasks);
+        taskCount++;
+    }
+
+    public static void addTodo(String input, Task[] tasks) {
+        int dividePos = input.trim().indexOf(" ");
+        String taskName = input.trim().substring(dividePos);
+        tasks[taskCount] = new Todo(taskName);
+    }
+
+    public static void addDeadlineOrEvent(String input, Task[] tasks) {
+        int dividePos = input.trim().indexOf(" ");
+        int timePos = input.trim().indexOf("/");
+        String taskType = input.trim().substring(0, dividePos).toLowerCase();
+        String taskName = input.trim().substring(dividePos, timePos);
+        String end = input.trim().substring(timePos + 3);
+        if (taskType.equalsIgnoreCase("deadline")) {
+            tasks[taskCount] = new Deadline(taskName, end);
+        } else if (taskType.equalsIgnoreCase("event")) {
+            tasks[taskCount] = new Event(taskName, end);
+        }
+    }
+
+    public static void requestList(Task[] tasks) {
+        printHorizontalLine();
+        if (taskCount == 1) {
+            System.out.println("There are no tasks!");
+            printHorizontalLine();
+            return;
+        }
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 1; i < taskCount; i++) {
+            System.out.println(i + "." + tasks[i]);
+        }
+        printHorizontalLine();
+    }
+
+    public static void inputBye() {
```

Comment:
Maybe this method could be named more explicitly such as printFarewellMessage? In my opinion, inputBye does not accurately describe the function of this method. I noticed the same issue in several other places too.